pinot-core/src/main/java/org/apache/pinot/core/accounting/PerQueryCPUMemAccountantFactory.java

```diff
@@ -786,6 +811,12 @@ public void runOnce() {
           reapFinishedTasks();
           if (_triggeringLevel.ordinal() > TriggeringLevel.Normal.ordinal()) {
             _aggregatedUsagePerActiveQuery = getQueryResourcesImpl();
+            _aggregatedUsagePerActiveQuery.values().stream()
+                .filter(stats -> !_cancelSentQueries.contains(stats.getQueryId()))
```

Comment:
we should kill the query with a global id or group queries together for the memory.